src/UnderlineNav/UnderlineNav.tsx

```diff
@@ -0,0 +1,173 @@
+import React, {useRef, forwardRef, useCallback, useState, useLayoutEffect, MutableRefObject} from 'react'
+import Box from '../Box'
+import {merge, SxProp, BetterSystemStyleObject} from '../sx'
+import {UnderlineNavContext} from './UnderlineNavContext'
+import {ActionMenu} from '../ActionMenu'
+import {ActionList} from '../ActionList'
+import {useResizeObserver} from '../hooks/useResizeObserver'
+
+/*
+Todo:
+1. Convert context to have array of all items = done
+2. Convert UnderlineNav.Link to ActionList.Item - done
+3. Calculate combined width of items to be 50% of total width available - done
+4. fix the dropdown button - done
+5. figure out horizontal scroll - done
+6. Overflow on resize - done
+7. pass typescript - done
+8. Performance is wack! Resize Observer exceeds limit
+9. Test for long strings not wrapping sometimes.
+10. Test for mobile device support.
+11. Add api for Overflow button?
+12. Do we need icons in overflow dropdown? Currently not supported.
+12. Write out a proper storybook with tab selection using this nav.
+13. Write up documentation
+14. Demo video.
+*/
+
+type Overflow = 'auto' | 'menu' | 'scroll'
+type ChildWidthArray = Array<{width: number}>
+type ResponsiveProps = {
+  items: Array<React.ReactElement>
+  actions: Array<React.ReactElement>
+}
+const overflowEffect = (
+  newRef: MutableRefObject<HTMLElement>,
+  childArray: Array<React.ReactElement>,
+  childWidthArray: ChildWidthArray,
+  callback: (props: ResponsiveProps) => void
+) => {
+  const domRect = newRef.current.getBoundingClientRect()
```

Comment:
Can't we get the dimensions from `useResizeObserver`? This would help us avoid the reflow that gets triggered when you call `.getBoundingClientRect()`

Reply:
Ah! this is good catch. I'll try to change the impl to use it. I implemented this part first and then wrapped it with the observer callback. So I didn't think of using it.

Though for first pass I'd still need this. So I'll do a conditional of if I have resizeObserver props, I'll use them, if not no.

Reply:
Any luck on this?